app/models/character_class.rb

```diff
@@ -1,3 +1,14 @@
 class CharacterClass < ApplicationRecord
   has_many :virtues #TODO define the inverse_of option for our associations in models
+  belongs_to :hero
+  has_many :states
+  def self.select_values
+    CharacterClass.all.map { |character_class| [character_class.name, character_class.id, { data: { url: data_url(character_class) }}]}
+  end
+
+  private
+
+  def self.data_url(character_class)
+    Rails.application.routes.url_helpers.character_class_virtues_path(character_class, format: :json)
```

Comment:
Sole fact that you have to call `Rails` constant in order to achieve something should ring an alarm. Here: the general rule of thumb is that you should not ever need to call url helpers from the models context